### ISSUE TITLE
What should I be careful of when working with update-freq 1?

### ISSUE THREAD

**Issue by JJumSSu** [pytorch/fairseq#1334]
Hi, I'm currently training transformer-base model on IWSLT17 dataset with ```update-freq``` 1. I intentionally set ```update-freq``` as 1 to fairly compare with other papers. Though when working with ```update-freq``` 1, the loss doesn't seem to go down as expected and training fails(goes up after 10~20 epochs). I tried changing learning rate from 0.0001 to 0.0006 with no success. Any other things that I should be cautious about?
Thank you!!

**Comment by huihuifan** [comment]
the batch size (affected by update freq) and learning rate need to be adjusted together, as they are correlated. please try reducing the learning rate if you are going to train with smaller batch size. Note update freq being 1 doesn't necessarily equate to the setting of other papers, please check and compare batch sizes and number of tokens in each batch
